pynest/setup.py.in

```diff
@@ -32,7 +32,8 @@ setup(
               'nest.tests.test_spatial', 'nest.lib'],
     install_requires=['numpy', 'scipy'],
     extras_require={
-        'test': ['nose', 'matplotlib']
+        'test': ['nose', 'matplotlib'],
```

Comment:
I think this line can go, since #2108 removed the `test` directory from PyNEST. Correct me if I'm wrong.

Reply:
The syntax is such that you can install optional requirements by invoking e.g. ``pip install nest-simulator[test]``, where ``test`` corresponds to this dictionary key.

Reply:
But there are no tests anymore in PyNEST. They are all in the general `testsuite` directory now.

Reply:
Where can one find pip installable test requirements for the NEST project then? We could get on board of the `pyproject.toml` file and follow PEP621: https://www.python.org/dev/peps/pep-0631/

Reply:
@terhorstd

Reply:
@jougs, the word `'test'` is just the label for the pip installable feature, not the directory. I think it makes perfect sense to have
* `pip install nest-simulator` to install without requirements for running the test suite, and
* `pip install nest-simulator[test]` which installs NEST and runs the test suite (independent of it's location).

Your point about placing the tests under `pynest/test` vs. `testsuite` seems to be pointing at a distinction we didn't do till now: testing NEST functionality through the pynest interface (similar to the SLI tests), versus testing functionality of pynest itself, i.e. the Python adapter classes, collections, etc.

It is currently not possible to run all required tests without having pynest and SLI installed, given the amount of tests done thrhough either interface. Python is also a hard requirement since NEST 3.0 and I strongly object to using `pip install` to install NEST without the pynest part.

Concluding, I'd say the line has to stay in, since those are the extra requirements for running the tests. No need to pull those in if you do not want to run the test suite anyway.

Just my ignorance:
* shouldn't `matplotlib` be an extra requirement for the examples instead of the test suite?
* why is there still `nose` instead of `pytest`, didn't we change this quite a while ago?